### PR TITLE
Add e2e Test Description to User-Agent

### PR DESCRIPTION
**Pull request by hh**
This PR appends detail e2e test line number and test details to the User-Agent.
It's needed for https://github.com/kubernetes/community/pull/2363 and Conformance Data gathering.

This can be easily tested with any 1.12.0-alpha or later that is configured for audit-logging.

Here is an example using the test-infra/kubetest dind from https://github.com/kubernetes/test-infra/pull/9061

```
  cd ~/go/src/k8s.io/kubernetes/
  go get -u k8s.io/test-infra/kubetest
  kubetest --build=dind
  kubetest --up --deployment=dind
  export KUBERNETES_CONFORMANCE_TEST=y
  export KUBECONFIG=$(ls -rt /tmp/k8s-dind-kubecfg-* | tail -1)
  export DIND_K8S_DATA=$(ls -drt /tmp/dind-k8* | tail -1)
  make -j 8 GOGCFLAGS="-N -l -v" WHAT=test/e2e/e2e.test
  ./_output/local/bin/linux/amd64/e2e.test \
    --ginkgo.focus=\[Conformance\] \
    --ginkgo.seed=1436380640 \
    --v=2 \
    --ginkgo.skip='\[Slow\]|\[Serial\]|\[Disruptive\]|\[Flaky\]|\[Feature:.+\]|\[HPA\]|Dashboard|Services.*functioning.*NodePort' 
  cp $DIND_KCS_DATA/audit/audit.log .
```

```
e2e.test/v1.12.0 (linux/amd64) kubernetes/94c2c6c -- k8s.io/kubernetes/test/e2e/kubectl/portforward.go:496 -- [sig-cli] Kubectl Port forwarding [k8s.io] With a server listening on localhost [k8s.io] that expects NO client request should support a client that connects, sends DATA, and disconnects
e2e.test/v1.12.0 (linux/amd64) kubernetes/94c2c6c -- k8s.io/kubernetes/test/e2e/network/dns.go:158 -- [sig-network] DNS should provide DNS for ExternalName services
e2e.test/v1.12.0 (linux/amd64) kubernetes/94c2c6c -- k8s.io/kubernetes/test/e2e/framework/framework.go:710 -- [sig-cli] Kubectl client [k8s.io] Kubectl cluster-info should check if Kubernetes master services is included in cluster-info  [Conformance]
e2e.test/v1.12.0 (linux/amd64) kubernetes/94c2c6c -- k8s.io/kubernetes/test/e2e/storage/persistent_volumes-local.go:562 -- [sig-storage] PersistentVolumes-local  StatefulSet with pod affinity should use volumes on one node when pod has affinity
```

Sample log file from e2e run:

[audit.log.gz](https://github.com/kubernetes/kubernetes/files/2298944/audit.log.gz)

### CODE REVIEW

**Comment by hh** [comment]
For tests that are not conformance tests, the line numbers show up fine:

```
k8s.io/kubernetes/test/e2e/kubectl/portforward.go:496 -- [sig-cli] Kubectl Port forwarding [k8s.io] With a server listening on localhost [k8s.io] that expects NO client request should support a client that connects, sends DATA, and disconnects
k8s.io/kubernetes/test/e2e/network/dns.go:158 -- [sig-network] DNS should provide DNS for ExternalName services
k8s.io/kubernetes/test/e2e/framework/framework.go:710 -- [sig-cli] Kubectl client [k8s.io] Kubectl cluster-info should check if Kubernetes master services is included in cluster-info  [Conformance]
k8s.io/kubernetes/test/e2e/storage/persistent_volumes-local.go:562 -- [sig-storage] PersistentVolumes-local  StatefulSet with pod affinity should use volumes on one node when pod has affinity
```

However when we use the ConformanceIt() wrapper, we lose this context, as the source is always 'k8s.io/kubernetes/test/e2e/framework/framework.go:757' :

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -155,6 +155,27 @@ func (f *Framework) BeforeEach() {
...
+			}
+			// in case of fileName:LineNumber ~ test/e2e/framework/framework.go:757
+			// it's hitting the ConformanceIt() wrapper....
+			// We'll need a work around to find conformance test sources
+			componentTexts := strings.Join(testDesc.ComponentTexts, " ")
+			config.UserAgent = fmt.Sprintf(
+				"%v -- %v:%v -- %v",
+				rest.DefaultKubernetesUserAgent(),
+				fileName,
+				testDesc.LineNumber,
```
Comment (spiffxp) [inline test/e2e/framework/framework.go]: I'm not sure how interesting or useful it is to go as low level as file:line if we have access to the test description

**Comment by spiffxp** [comment]
/release-note-none
/hold
for comment... I'm interested in getting this merged but I think I'd prefer removing the additional complexity of line numbers / files

**Comment by spiffxp** [comment]
/approve
but no /lgtm due to above

**Comment by liggitt** [comment]
> for comment... I'm interested in getting this merged but I think I'd prefer removing the additional complexity of line numbers / files

agree. api call by test seems useful

**Comment by hh** [comment]
I removed file / line numbers.... henceforth tests will be known by `strings.Join(CurrentGinkoTestDesctription.ComponentTexts, " ")`

**Comment by dims** [comment]
LGTM, please squash the commits and fix the description to drop the Filename:Linenumber and we should be good.

**Comment by hh** [comment]
Squashed and pushed.

File: test/e2e/framework/framework.go [hunk 2]
```diff
@@ -155,6 +155,16 @@ func (f *Framework) BeforeEach() {
 	if f.ClientSet == nil {
 		By("Creating a kubernetes client")
 		config, err := LoadConfig()
+		testDesc := CurrentGinkgoTestDescription()
+		switch {
+		case len(testDesc.ComponentTexts) > 0:
```
Comment (mikedanese) [inline test/e2e/framework/framework.go]: use an if

Comment (timothysc) [inline test/e2e/framework/framework.go]: can you just switch this to an if statement?  The switch case for a case of one makes little sense imo.

**Review by timothysc — CHANGES_REQUESTED** [review]

**Comment by spiffxp** [comment]
/test pull-kubernetes-kubemark-e2e-gce-big
submit queue claims this test is failing but I see all green here https://k8s-gubernator.appspot.com/pr/67550

**Pull request merged**